api/src/main/java/org/openmrs/module/commonreports/reports/FamilyPlanningReportManager.java

```diff
@@ -0,0 +1,177 @@
+		designProperties.put("PC.label", MessageUtil.translate("commonreports.report.familyPlanning.PC.label"));
+		designProperties.put("PP.label", MessageUtil.translate("commonreports.report.familyPlanning.PP.label"));
+		designProperties.put("depo.label", MessageUtil.translate("commonreports.report.familyPlanning.depo.label"));
+		designProperties.put("implant.label", MessageUtil.translate("commonreports.report.familyPlanning.implant.label"));
+		designProperties.put("condoms.label", MessageUtil.translate("commonreports.report.familyPlanning.condoms.label"));
+		designProperties.put("total.label", MessageUtil.translate("commonreports.report.familyPlanning.total.label"));
+		
+		reportDesign.setProperties(designProperties);
+		return Arrays.asList(reportDesign);
+	}
+	
+	private String applyMetadataReplacements(String rawSql) {
+		String s = rawSql
+		        .replace(":FPAdministred",
+		            inizService.getConceptFromKey("report.familyPlanning.FPAdministred").getConceptId() + "")
+		        .replace(":familyPlanning",
+		            inizService.getConceptFromKey("report.familyPlanning.familyPlanning").getConceptId() + "")
+		        .replace(":femaleLT25",
+		            "person.gender = 'F' AND round(DATEDIFF(obs.obs_datetime, person.birthdate)/365.25, 1) < 25")
+		        .replace(":femaleGT25",
+		            "person.gender = 'F' AND round(DATEDIFF(obs.obs_datetime, person.birthdate)/365.25, 1) >= 25")
+		        .replace(":maleLT25",
+		            "person.gender = 'M' AND round(DATEDIFF(obs.obs_datetime, person.birthdate)/365.25, 1) < 25")
+		        .replace(":maleGT25",
+		            "person.gender = 'M' AND round(DATEDIFF(obs.obs_datetime, person.birthdate)/365.25, 1) >= 25")
+		        
+		        .replace(":typeOfUser",
+		            inizService.getConceptFromKey("report.familyPlanning.typeOfUser").getConceptId() + "")
+		        .replace(":new", inizService.getConceptFromKey("report.familyPlanning.new").getConceptId() + "")
+		        .replace(":existent", inizService.getConceptFromKey("report.familyPlanning.existent").getConceptId() + "")
+		        
+		        .replace(":microgynon",
+		            inizService.getConceptFromKey("report.familyPlanning.microgynon").getConceptId() + "")
+		        
+		        .replace(":microlut", inizService.getConceptFromKey("report.familyPlanning.microlut").getConceptId() + "")
+		        
+		        .replace(":depoProveraInjection",
+		            inizService.getConceptFromKey("report.familyPlanning.depoProveraInjection").getConceptId() + "")
+		        .replace(":jadel", inizService.getConceptFromKey("report.familyPlanning.jadel").getConceptId() + "")
+		        .replace(":condom", inizService.getConceptFromKey("report.familyPlanning.condom").getConceptId() + "")
+		        .replace(":FPObsGroupId",
+		            "obs_group_id IN (select obs_group_id from obs where obs_group_id IN (SELECT obs_id FROM obs");
```

Comment:
This would rather stay in the sql query since the groupId itself cannot be determined from here.

Reply:
Done